hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/container/common/statemachine/container/ClusterContainersUtil.java

```diff
@@ -0,0 +1,146 @@
+      throws IOException {
+    Preconditions.checkArgument(key instanceof OzoneKeyDetails);
+    long containerID = ((OzoneKeyDetails) key).getOzoneKeyLocations().get(0)
+        .getContainerID();
+    long localID = ((OzoneKeyDetails) key).getOzoneKeyLocations().get(0)
+        .getLocalID();
+    // From the containerData, get the block iterator for all the blocks in
+    // the container.
+    KeyValueContainerData containerData =
+        (KeyValueContainerData) container.getContainerData();
+    try (DBHandle db = BlockUtils.getDB(containerData, cluster.getConf());
+         BlockIterator<BlockData> keyValueBlockIterator =
+             db.getStore().getBlockIterator(containerID)) {
+      // Find the block corresponding to the key we put. We use the localID of
+      // the BlockData to identify out key.
+      BlockData blockData = null;
+      while (keyValueBlockIterator.hasNext()) {
+        blockData = keyValueBlockIterator.nextBlock();
+        if (blockData.getBlockID().getLocalID() == localID) {
+          break;
+        }
+      }
+      assertNotNull(blockData, "Block not found");
+
+      // Get the location of the chunk file
+      String containreBaseDir =
+          container.getContainerData().getVolume().getHddsRootDir().getPath();
+      File chunksLocationPath = KeyValueContainerLocationUtil
+          .getChunksLocationPath(containreBaseDir, cluster.getClusterId(), containerID);
+      return chunksLocationPath;
+    }
+  }
+
+  /**
+   * Corrupt the chunk backing the key in a mini cluster.
+   * @param cluster a mini ozone cluster object.
+   * @param container a container object.
+   * @param key an OzoneKey object.
+   * @throws IOException
+   */
+  public static void corruptData(MiniOzoneCluster cluster, Container container, OzoneKey key)
+      throws IOException {
+    File chunksLocationPath = getChunksLocationPath(cluster, container, key);
+    byte[] corruptData = "corrupted data".getBytes(UTF_8);
+    // Corrupt the contents of chunk files
+    for (File file : FileUtils.listFiles(chunksLocationPath, null, false)) {
+      FileUtils.writeByteArrayToFile(file, corruptData);
+    }
+  }
+
+  /**
+   * Inspect and verify if chunk backing the key in a mini cluster is the same as the string.
+   * @param cluster a mini ozone cluster object.
+   * @param container a container object.
+   * @param key an OzoneKey object.
+   * @throws IOException
+   */
+  public static void verifyOnDiskData(MiniOzoneCluster cluster, Container container, OzoneKey key, String data)
+      throws IOException {
+    File chunksLocationPath = getChunksLocationPath(cluster, container, key);
+    for (File file : FileUtils.listFiles(chunksLocationPath, null, false)) {
+      String chunkOnDisk = FileUtils.readFileToString(file, Charset.defaultCharset());
+      if (!data.equals(chunkOnDisk)) {
+        throw new CorruptedFileException(file, " does not match the source.");
```

Comment:
Can we just make this return a boolean if the data does not match? Current usage in `TestOzoneAtRestEncryption` is kind of confusing since it appears to be checking for "corruption" when actually it just wants the data on disk to be different than the input.